ios/RNSScreenStack.mm

```diff
@@ -1175,7 +1171,11 @@ - (void)unmountChildComponentView:(UIView<RCTComponentViewProtocol> *)childCompo
 
 - (void)takeSnapshot
 {
-  _snapshot = [_controller.visibleViewController.view snapshotViewAfterScreenUpdates:NO];
+  if (_presentedModals.count < 2) {
+    _snapshot = [_controller.visibleViewController.view snapshotViewAfterScreenUpdates:NO];
+  } else {
+    _snapshot = [[_presentedModals.lastObject view] snapshotViewAfterScreenUpdates:NO];
+  }
```

Comment:
Yeah, I see. 
There, most likely, still be an issue when we dismiss a modal from different stack. I believe you can test this on `Test1829` (if not present in `FabricTestExample` it is in `TestsExample`).

Reply:
Yep, it does not work in such case, do we want to handle it right now?

Reply:
I think we won't be really able to come up with anything different that a heuristic solution before we do proper modal presenting logic refactor.

Let's proceed now, I'll create note on our task board so that this is not forgotten.